### PR TITLE
feat: util to serve images locally or via remote url

### LINKED ISSUE #22004: [Task]: Bundle and serve images locally for air-gapped instances.
### Is there an existing issue for this?

- [X] I have searched the existing issues

### SubTasks

_No response_

### PR DESCRIPTION
**Pull request by berzerkeer**
## Description

- On air-gapped instances we can't fetch appsmith assets from S3, that will result in broken which is not desirable. 
- So this adds a script and util function which searches the client and server codebase for the assets url and downloads the image and puts it in the `public` folder so that the browser can access those even in an airgapped instance since the assets are being served locally.

> Way to serve assets locally.

Fixes #22004 
> if no issue exists, please create an issue and ask the maintainers about this first

Media
https://vdqm24wed6.vmaker.com/record/IquS90WbWgS1I0bz - blocked certain api routes from getting called on airgap

## Type of change

> Please delete options that are not relevant.

- New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

- Manual

### Dev activity
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

### QA activity:
- [ ] Test plan has been approved by relevant developers
- [ ] Test plan has been peer reviewed by QA
- [ ] Cypress test cases have been added and approved by either SDET or manual QA
- [ ] Organized project review call with relevant stakeholders after Round 1/2 of QA
- [ ] Added Test Plan Approved label after reveiwing all Cypress test

### CODE REVIEW

File: app/client/src/pages/Editor/APIEditor/ProviderTemplates.tsx [hunk 1]
```diff
@@ -318,7 +318,7 @@ class ProviderTemplates extends React.Component<ProviderTemplatesProps> {
             {" Back"}
           </span>
           <br />
-
+          {/* TODO: Check image URls */}
```
Comment (dipyamanbiswas07) [inline app/client/src/pages/Editor/APIEditor/ProviderTemplates.tsx:321]: Could you add your name to the todo comment.

File: app/client/src/ce/utils/airgapHelpers.tsx [hunk 2]
```diff
@@ -0,0 +1,7 @@
+export const getAssetUrl = (src = "") => {
+  return src;
+};
+
+export const isAirgapped = () => {
```
Comment (dipyamanbiswas07) [inline app/client/src/ce/utils/airgapHelpers.tsx:5]: One question around this. 

Since this is going to be used in a lot of places, could we do the computation once and store it? either in redux or as a static class property?

**Review by dipyamanbiswas07 — DISMISSED** [review]

File: app/client/src/ce/utils/airgapHelpers.tsx (hunk 2 above)
Comment (berzerkeer) [reply app/client/src/ce/utils/airgapHelpers.tsx:5]: AFAIK this won't make any difference since there is no computation here, it just returns false here and on EE also it actually just returns what we have in the window object ( so basically like redux store ) and these values are set when we initialize the app and not going to change afterwards so we don't really need state management stuff. If we do it in redux there will be a lot of boilerplate code. So I guess what we have here is jut fine. Let me know if this makes sense , I could be wrong. Thanks !

File: app/client/src/pages/Editor/APIEditor/ProviderTemplates.tsx (hunk 1 above)
Comment (berzerkeer) [reply app/client/src/pages/Editor/APIEditor/ProviderTemplates.tsx]: They are not needed anymore, removed it.

File: app/client/src/ce/utils/airgapHelpers.tsx (hunk 2 above)
Comment (dipyamanbiswas07) [reply app/client/src/ce/utils/airgapHelpers.tsx:5]: Got it. Since there are no computations required, we are good.

**Review by dipyamanbiswas07 — APPROVED** [review]

**Review by dipyamanbiswas07 — APPROVED** [review]

File: app/client/src/pages/Editor/QueryEditor/EditorJSONtoForm.tsx [hunk 3]
```diff
@@ -576,13 +577,15 @@ export function EditorJSONtoForm(props: Props) {
   }
 
   function SingleValue(props: SingleValueProps<OptionTypeBase>) {
+    {
```
Comment (ankitakinger) [inline app/client/src/pages/Editor/QueryEditor/EditorJSONtoForm.tsx:580]: why is this needed?

File: app/client/src/constants/ImagesURL.ts [hunk 4]
```diff
@@ -1,5 +1,7 @@
 import { ASSETS_CDN_URL } from "./ThirdPartyConstants";
+import { getAssetUrl } from "@appsmith/utils/airgapHelpers";
+
 export const getInfoThumbnail = () =>
-  `${ASSETS_CDN_URL}/crud/crud_info_thumbnail.png`;
-export const getInfoImage = (): string =>
-  `${ASSETS_CDN_URL}/crud/working-flow-chart.png`;
+  getAssetUrl(`${ASSETS_CDN_URL}/crud/crud_info_thumbnail.png`);
+export const getInfoImage = () =>
```
Comment (ankitakinger) [inline app/client/src/constants/ImagesURL.ts:6]: will `getInfoImage` return type not be a string anymore?

File: app/client/src/pages/Editor/QueryEditor/EditorJSONtoForm.tsx (hunk 3 above)
Comment (berzerkeer) [reply app/client/src/pages/Editor/QueryEditor/EditorJSONtoForm.tsx:580]: Oh this is just a typo, I added a console log to check the props ,only the console log removed and not the braces , I will remove this now. Good catch, Thanks !

File: app/client/src/constants/ImagesURL.ts (hunk 4 above)
Comment (berzerkeer) [reply app/client/src/constants/ImagesURL.ts:6]: It will always be string , will add it back. I must've removed it when testing.

**Review by ankitakinger — APPROVED** [review]

**Review by ankitakinger — APPROVED** [review]

File: app/client/src/selectors/editorSelectors.tsx [hunk 5]
```diff
@@ -342,10 +343,9 @@ export const getWidgetCards = createSelector(
   getWidgetConfigs,
   getIsAutoLayout,
   (widgetConfigs: WidgetConfigReducerState, isAutoLayout: boolean) => {
-    const cards = Object.values(widgetConfigs.config).filter(
-      (config) => !config.hideCard,
-    );
-
+    const cards = Object.values(widgetConfigs.config).filter((config) => {
+      return isAirgapped() ? config.widgetName !== "Map" : !config.hideCard;
```
Comment (iamenrique) [inline app/client/src/selectors/editorSelectors.tsx:347]: This means that air-gapped instances allow "hidden cards" to be displayed? 
image

Comment (berzerkeer) [reply app/client/src/selectors/editorSelectors.tsx:347]: Nope, it was a bug. This has been fixed in a different PR. https://github.com/appsmithorg/appsmith/pull/22440
